include/ginkgo/core/base/lin_op.hpp

```diff
@@ -622,6 +622,25 @@ class DiagonalExtractable {
         const = 0;
 };
 
+
+/**
+ * The absolute of a LinOp implementing this interface can be computed.
+ * get_absolute get a new LinOp which contains absolute values from a LinOp.
+ *
+ *
+ * @ingroup LinOp
+ */
+class AbsoluteComputable {
```

Comment:
Can we make this an interface ? `EnableAbsoluteComputation<ConcreteType>`. I would prefer this because then it can return the `ConcreteType` rather than a `LinOp` and we dont have to use `as` and dynamic casts everywhere.

Reply:
You mean a `mixin`. Yes, that does make sense.

Reply:
I thought it before, but it required the result type when the object is LinOp and it need to do dynamic_cast before using it.
```
// A is LinOp* currently
auto abs_obj = gko::as<AbsoluteComputable<csr<rmc_t, idx_t>>>(A);
auto abs_res = abs_obj->get_absolute()
```